src/features/targeting/targeting-api.js

```diff
@@ -1,23 +1,15 @@
 // @flow
-import type { MessageHeader } from './types';
-
 /**
  * TargetingApi is provided by TargetingContext. It is provided on a per message basis.
  * const { getTargetingApi } = useContext(TargetingContext);
  * const targetingApi = getTargetingApi(messageName)
  * will return the TargetingApi for the given message. See README.
  */
 export type TargetingApi = {|
-    onClosed: () => void,
-    // call to onSeen will cause the message seen count to increment
+    // call onSeen after message is shown
     onSeen: () => void,
-    // call to onClosed will cause shouldShow to be false for the rest of the session, and
-    // message close count to increment.
+    // return true if message is eligible to show and not yet closed
     shouldShow: () => boolean,
+    // call toClose just before message should be closed, to cause shouldShow to return false
+    toClose: () => void,
```

Comment:
sorry can we make this `onClose` (i'm so sorry that my comment previously was inaccurate).